File: chroma-league/src/test/java/net/booone/chromaleague/hud/parts/resource/ManaBarTest.java
package net.booone.chromaleague.hud.parts.resource;

import net.booone.chromaleague.IntSteps;
import net.booone.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

class ManaBarTest extends AbstractResourceTest {
    @Test
    void testManaBar() {
        final IntSteps intSteps = new IntSteps(0, 100, 5);
        final ManaBar manaBar = new ManaBar();

        new AnimationTester()
                .withBeforeIterationAction(i -> mockResource(intSteps.nextInt(), 100))
                .testAnimation(manaBar, 40);
    }
}

File: test/hud/parts/resource/ManaBar.test.ts
import {describe, test} from 'vitest';
import {IntSteps} from '../../../IntSteps.js';
import {AnimationTester} from '../../../AnimationTester.js';
import {ManaBar} from '../../../../src/hud/parts/resource/ManaBar.js';
import {mockResource, setupResourceTest} from './AbstractResourceTest.js';

/** Integration test for mana resource bar on actual hardware. */
describe('Mana Integration', () => {
    setupResourceTest();

    test('testManaBar', async () => {
        const intSteps = new IntSteps(0, 100, 5);
        const manaBar = new ManaBar();

        await new AnimationTester()
            .withBeforeIterationAction(() => mockResource(intSteps.nextInt(), 100))
            .testAnimation(manaBar, 40);
    }, 30000);
});
